azurelinuxagent/ga/exthandlers.py

```diff
@@ -191,21 +191,28 @@ def __init__(self):
         self.log_process = False
 
         self.report_status_error_state = ErrorState()
+        self.get_artifact_error_state = ErrorState(min_timedelta=datetime.timedelta(minutes=5))
 
     def run(self):
         self.ext_handlers, etag = None, None
         try:
             self.protocol = self.protocol_util.get_protocol()
             self.ext_handlers, etag = self.protocol.get_ext_handlers()
+            self.get_artifact_error_state.reset()
         except Exception as e:
-            msg = u"Exception retrieving extension handlers: {0}".format(
-                ustr(e))
-            logger.warn(msg)
-            add_event(AGENT_NAME,
-                      version=CURRENT_VERSION,
-                      op=WALAEventOperation.ExtensionProcessing,
-                      is_success=False,
-                      message=msg)
+            msg = u"Exception retrieving extension handlers: {0}".format(ustr(e))
+            self.get_artifact_error_state.incr()
+
+            if self.get_artifact_error_state.is_triggered():
+                add_event(AGENT_NAME,
+                          version=CURRENT_VERSION,
+                          op=WALAEventOperation.GetArtifactExtended,
+                          is_success=False,
+                          message="Failed to get extension artifact for over "
+                                  "{0): {1}".format(self.get_artifact_error_state.min_timedelta, msg))
+                self.get_artifact_error_state.reset()
+            else:
+                logger.warn(msg)
```

Comment:
If we're going to retry, should this be verbose?  When retries have been exhausted we can log an error.

Reply:
We do retry, but any failure is still abnormal. If we change to verbose, we would lose this detail, since we hardly ever get verbose logs. This exception handler is high up enough that I feel it's worth keeping the detail at the expense of potentially cluttering the logs.